### PR TITLE
Fix/initialize markets

### PR DESCRIPTION
**Pull request by alexeckert**
Closes https://github.com/eprbell/dali-rp2/issues/161

### CODE REVIEW

File: src/dali/plugin/input/rest/kraken.py [hunk 1]
```diff
@@ -122,8 +122,13 @@ def _initialize_client(self) -> kraken:
 
     def _initialize_markets(self) -> None:
         self._client.load_markets()
-        self._client.markets_by_id.update({"BSVUSD": {_ID: "BSVUSD", _BASE_ID: "BSV", _BASE: "BSV", _QUOTE: "USD"}})
+        # Convert Dict[any, List[Dict]] to Dict[any, Dict]
+        for key, value in self._client.markets_by_id.items():
+            if len(value) != 1:
+                raise RP2RuntimeError("List of markets can't be converted to dictionary, since list does not contain exaxtly one market entry.")
```
Comment (eprbell) [inline src/dali/plugin/input/rest/kraken.py:128]: Typo: exactly

**Review by eprbell — CHANGES_REQUESTED** [review]
Other than a typo and some test failures, this LGTM (but I don't have a Kraken account). @macanudo527, can you also review this?

**Review by eprbell — CHANGES_REQUESTED** [review]
There's still a static analysis failure.

**Comment by alexeckert** [comment]
> There's still a static analysis failure.



> There's still a static analysis failure.

Yes, thanks the pylinted tests didn't like the name of the fixture. Ran pylint locally now, as it is not part of the pre-commit hook.

**Comment by eprbell** [comment]
> > There's still a static analysis failure.
> 
> Yes, thanks the pylinted tests didn't like the name of the fixture. Ran pylint locally now, as it is not part of the pre-commit hook.

No worries. There's still some mypy errors. You can run mypy locally with `https://github.com/eprbell/dali-rp2/blob/main/README.dev.md#development-workflow` and you can also run all the other CI test locally (see https://github.com/eprbell/dali-rp2/blob/main/README.dev.md#development-workflow).

**Comment by macanudo527** [comment]
It seems like #198 addresses this more robustly. Is it all right to close this PR?

**Comment by eprbell** [comment]
I tried to merge but there is one more small static analysis problem: once that's fixed this is good to go.

**Comment by alexeckert** [comment]
@macanudo527 Sure thanks, still had to fix the typing issue any way.

**Pull request closed without merging**